Model/Product/Collection.php

```diff
@@ -36,15 +36,19 @@
 
 namespace Nosto\Tagging\Model\Product;
 
+use Exception;
+use Magento\Catalog\Model\Product;
 use Magento\Catalog\Model\Product\Visibility as ProductVisibility;
+use Magento\Catalog\Model\ResourceModel\Product\Collection as MageProductCollection;
 use Magento\Catalog\Model\ResourceModel\Product\CollectionFactory as ProductCollectionFactory;
 use Magento\Sales\Api\Data\EntityInterface;
 use Magento\Store\Model\Store;
 use Nosto\NostoException;
 use Nosto\Object\Product\ProductCollection;
-use Nosto\Tagging\Model\Service\Product as NostoProductService;
+use Nosto\Tagging\Model\Service\ProductServiceInterface as NostoProductService;
```

Comment:
Why alias this? I'd say it's more readable and consistent with other interface injections if you leave it as `ProductServiceInterface`